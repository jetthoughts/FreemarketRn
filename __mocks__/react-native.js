const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
