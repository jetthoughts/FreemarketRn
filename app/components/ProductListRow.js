import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View, Image } from 'react-native';

import colors from '../constants/colors';

const ident = 10;
const styles = StyleSheet.create({
  row: {
    margin: ident,
    backgroundColor: colors.productBackground,
    borderRadius: 5,
    flexDirection: 'row',
    padding: ident,
  },
  image: {
    width: 100,
    height: 100,
  },
  descriptionBlock: {
    paddingLeft: ident,
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  name: {
    color: colors.textDark,
    fontSize: 15,
  },
  numbersBlock: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  timeBlock: {

  },
  timeTitle: {
    fontSize: 10,
  },
  timeValue: {
    color: colors.textAttention,
    fontSize: 15,
  },
  priceBlock: {
    backgroundColor: colors.priceBackground,
    width: 100,
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  price: {
    color: colors.textLight,
  },
});

const ProductListRow = ({ name, image, price }) => (
  <View style={styles.row}>
    <Image
      style={styles.image}
      source={{ uri: image }}
    />
    <View style={styles.descriptionBlock}>
      <Text style={styles.name}>{name}</Text>
      <View style={styles.numbersBlock}>
        <View style={styles.timeBlock}>
          <Text style={styles.timeTitle}>Time Left:</Text>
          <Text style={styles.timeValue}>17m 5s</Text>
        </View>
        <View style={styles.priceBlock}>
          <Text style={styles.price}>${price.toFixed(2)}</Text>
        </View>
      </View>
    </View>
  </View>
);

ProductListRow.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
};

export default ProductListRow;
