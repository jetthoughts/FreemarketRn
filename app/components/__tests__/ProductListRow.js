import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProductListRow from '../ProductListRow';

it('renders correctly', () => {
  const tree = renderer.create(
    <ProductListRow
      name="Item"
      image="http://i.ebayimg.com/images/g/64EAAOSwCQZZDJPO/s-l1600.jpg"
      price={10}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
