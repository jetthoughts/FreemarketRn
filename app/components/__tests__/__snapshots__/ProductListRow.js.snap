// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "flexDirection": "row",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "http://i.ebayimg.com/images/g/64EAAOSwCQZZDJPO/s-l1600.jpg",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "paddingLeft": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 15,
        }
      }
    >
      Item
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={Object {}}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
            }
          }
        >
          Time Left:
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "red",
              "fontSize": 15,
            }
          }
        >
          17m 5s
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "dimgray",
            "borderRadius": 5,
            "justifyContent": "center",
            "width": 100,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
            }
          }
        >
          $
          10.00
        </Text>
      </View>
    </View>
  </View>
</View>
`;
